src/lib/OS/SimOS.cc

```diff
@@ -223,5 +231,38 @@ void SimOS::createSpecialFileDirectory() const {
   SFdir.GenerateSFDir();
 }
 
+uint64_t SimOS::requestPageFrames(size_t size) {
+  return pageFrameAllocator_.allocate(size);
+}
+
+uint64_t SimOS::handleVAddrTranslation(uint64_t vaddr, uint64_t tid) {
+  auto process = processes_.find(0)->second;
+  uint64_t translation = process->pageTable_->translate(vaddr);
+  uint64_t faultCode = masks::faults::getFaultCode(translation);
+
+  // Return the translation if faultCode is not translate, any other faults
+  // will be handled further. Only page faults will be handled here.
+  if (faultCode != masks::faults::pagetable::TRANSLATE) return translation;
+
+  uint64_t addr = process->handlePageFault(vaddr, sendToMem_);
+  faultCode = masks::faults::getFaultCode(addr);
+
+  if (faultCode == masks::faults::pagetable::MAP) {
+    std::cerr << "[SimEng:SimOS] Failed to create mapping during PageFault "
+                 "caused by Vaddr: "
+              << vaddr << "( TID: " << tid << " )" << std::endl;
+    std::exit(1);
+  }
+
+  return addr;
+}
+
+VAddrTranslator SimOS::getVAddrTranslator() {
+  auto fn = [&, this](uint64_t vaddr, uint64_t pid) -> uint64_t {
+    return this->handleVAddrTranslation(vaddr, pid);
+  };
```

Comment:
Need only pass `[this]` and can remove `this->` from function body

Reply:
done